File: 20-valid-parentheses/20-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
    int i=-1;
    for(char ch:s) {
        if(ch == '(' || ch == '{' || ch == '[')
            s[++i] = ch;
        else if(i>-1 && ((ch=='}'&&s[i]=='{') || (ch==')'&&s[i]=='(') || (ch==']'&&s[i]=='[')))
            i--;
        else
            return false;
    }
    return i==-1;
}

};

File: 20-valid-parentheses/20-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        if(s.length()%2!=0)return false;
      Stack<Character>stack=new Stack<>();
       for(int i=0;i<s.length();i++){
           if(s.charAt(i)=='('||s.charAt(i)=='{'||s.charAt(i)=='[')
               stack.push(s.charAt(i));
           else{
               if(stack.isEmpty())return false;
               char c=stack.pop();
               if(s.charAt(i)==']'&&c!='['||
                  s.charAt(i)=='}'&&c!='{'||
                  s.charAt(i)==')'&&c!='(')
                   return false;  
           }
       }
        return stack.isEmpty();
    }
}